inbm/dispatcher-agent/dispatcher/aota/application_command.py

```diff
@@ -212,13 +246,16 @@ def __init__(self,  dispatcher_broker: DispatcherBroker,
         super().__init__(dispatcher_broker, parsed_manifest, dbs, update_logger=update_logger)
 
     def update(self):  # pragma: no cover
-        super().update()
+        super().update()        
         application_repo = self._download_package()
-        install_cmd = application_repo.get_repo_path() + "/" + self.resource if self.resource else ""
-        if ' ' in install_cmd or install_cmd.isspace():
-            logger.debug(f"INSTALL : {install_cmd}")
-            raise AotaError(f"File path cannot contain spaces - {install_cmd}")
-        base_command = f"/usr/bin/dpkg -i {install_cmd}"
+        path_to_pkg = application_repo.get_repo_path() + "/" + self.resource if self.resource else ""
+        if ' ' in path_to_pkg or path_to_pkg.isspace():
+            logger.debug(f"INSTALL : {path_to_pkg}")
+            raise AotaError(f"File path cannot contain spaces - {path_to_pkg}")
+
+        self.run_signature_check(path_to_file=path_to_pkg)
+
+        base_command = f"/usr/bin/dpkg -i {path_to_pkg}"
 
         is_docker_app = os.environ.get("container", False)
         if is_docker_app:
```

Comment:
**Suggestion:** To avoid hardcoding the reboot command, consider defining it as a class or module constant. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        REBOOT_BASE_COMMAND = "/sbin/reboot"
        command_prefix = DOCKER_CHROOT_PREFIX if os.environ.get("container", False) else ""
        reboot_command = f"{command_prefix}{REBOOT_BASE_COMMAND}"
```